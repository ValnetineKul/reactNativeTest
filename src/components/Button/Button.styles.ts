import { StyleSheet } from "react-native";
import {
  COLOR_CARD_BOX_SHADOW,
  COLOR_ERROR,
  COLOR_TEXT_BLUE,
  COLOR_TEXT_GRAY,
} from "../../theme";
import { flexHelper } from "../../utils";
import { FontWeights } from "../Typography";

export const styles = StyleSheet.create({
  button: {
    padding: 12,
    backgroundColor: COLOR_TEXT_BLUE,
    borderRadius: 4,
    shadowColor: COLOR_CARD_BOX_SHADOW,
    shadowOffset: { width: 2, height: 4 },
    shadowOpacity: 0.3,
    width: 125,
    height: 40,
  },
  fullWidth: {
    width: "100%",
  },
  title: {
    fontWeight: FontWeights.fontWeightMedium,
    textAlign: "center",
  },
  startAdorment: {
    marginRight: 16,
  },
  endAdorment: {
    marginLeft: 16,
  },
  flexProvider: {
    ...flexHelper({}),
  },
  blue: {
    backgroundColor: COLOR_TEXT_BLUE,
  },
  gray: {
    backgroundColor: COLOR_TEXT_GRAY,
  },
  red: {
    backgroundColor: COLOR_ERROR,
  },
});
