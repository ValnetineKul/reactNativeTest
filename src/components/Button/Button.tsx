import React, { PropsWithChildren, ReactElement } from "react";
import { TouchableOpacity, View, ViewStyle } from "react-native";
import { Typography } from "../Typography";
import { styles } from "./Button.styles";

export type ButtonProps = {
  fullWidth?: boolean;
  onPress?: () => void;
  style?: ViewStyle | ViewStyle[];
  title: string;
  startAdorment?: ReactElement;
  endAdorment?: ReactElement;
  color?: "blue" | "gray" | "red";
  disabled?: boolean;
};

export const Button = ({
  onPress,
  style,
  title,
  fullWidth,
  startAdorment,
  endAdorment,
  color = "blue",
  disabled,
}: PropsWithChildren<ButtonProps>) => {
  return (
    <TouchableOpacity
      style={[styles.button, styles[color], { ...(fullWidth && styles.fullWidth) }, style]}
      onPress={onPress}
      disabled={disabled}
    >
      <View style={[styles.flexProvider]}>
        {startAdorment && <View style={styles.startAdorment}>{startAdorment}</View>}
        <Typography variant="body1" color="white" style={styles.title}>
          {title}
        </Typography>
        {endAdorment && <View style={styles.endAdorment}>{endAdorment}</View>}
      </View>
    </TouchableOpacity>
  );
};
