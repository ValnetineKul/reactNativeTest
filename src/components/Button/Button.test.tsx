import React from "react";
import renderer from 'react-test-renderer';
import { render, screen } from "@testing-library/react-native";
import { Button } from "./Button";

describe("Button tests", () => {
  it("renders the correct message", () => {
    render(<Button title="Hello world" />);
    expect(screen.getByText("Hello world")).toBeTruthy();
  });

  it('renders correctly when there are no items', () => {
    const tree = renderer.create(<Button title={"Hello world"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
