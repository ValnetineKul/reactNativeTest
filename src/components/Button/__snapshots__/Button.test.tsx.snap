// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests renders correctly when there are no items 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#00A8F3",
      "borderRadius": 4,
      "height": 40,
      "opacity": 1,
      "padding": 12,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 4,
        "width": 2,
      },
      "shadowOpacity": 0.3,
      "width": 125,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Roboto",
          },
          {
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          {
            "color": "#FFFFFF",
          },
          {
            "fontWeight": "500",
            "textAlign": "center",
          },
        ]
      }
    >
      Hello world
    </Text>
  </View>
</View>
`;
